modules/social_features/social_follow_user/src/Service/SocialFollowUserHelper.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\social_follow_user\Service;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\flag\FlagServiceInterface;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\user\UserDataInterface;
+
+/**
+ * Defines the helper service.
+ */
+class SocialFollowUserHelper implements SocialFollowUserHelperInterface {
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The user data service.
+   *
+   * @var \Drupal\user\UserDataInterface
+   */
+  protected $userData;
+
+  /**
+   * The flag service.
+   *
+   * @var \Drupal\flag\FlagServiceInterface
+   */
+  protected $flagService;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    AccountProxyInterface $current_user,
+    UserDataInterface $user_data,
+    FlagServiceInterface $flag,
+    EntityTypeManagerInterface $entity_type_manager
+  ) {
+    $this->currentUser = $current_user;
+    $this->userData = $user_data;
+    $this->flagService = $flag;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isFollowingEnabled(ProfileInterface $profile): bool {
+    $following_enabled = TRUE;
+
+    // Check if disabled user following due to privacy settings.
+    if (!$this->getFollowingStatus((int) $profile->get('uid')->target_id)) {
+      $following_enabled = FALSE;
+
+      // Check if user already followed.
+      /** @var \Drupal\flag\FlagInterface $flag */
+      $flag = $this->flagService->getFlagById('follow_user');
+      // And display only "Unfollow" button because we should leave the ability
+      // to unfollow user.
+      if ($this->flagService->getFlagging($flag, $profile, $this->currentUser)) {
+        $following_enabled = TRUE;
+      }
+    }
+    return $following_enabled;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setFollowingStatus(int $uid, $status = TRUE): void {
+    $this->userData->set(
```

Comment:
@volodymyr-sydor I more like an approach with a new profile field instead of the user data. Since if we use the user data we need to use batch for hook_install to set the default value and I think it is not good https://github.com/goalgorilla/open_social/pull/2967/files#diff-d3d7b1d9964a18dba55138aef6c9585a65978162753ff93181c2709491e32289R38
But I also understand that many fields for the profile are also not well. So I want to hear your thoughts @navneet0693 @tbsiqueira

cc @ribel

Reply:
@Ressinel Yes, seems like it will be even easier if we create a new profile field, for example, `field_profile_following` then we don't need to update existing users & add any action for new accounts, etc.
@navneet0693 @tbsiqueira are you ok with that approach?

Reply:
@tbsiqueira 
The acceptance criteria are that `everyone that is already following the user should stay a follower` 
> Will my account still be listed under the "following users" overview from another user? 

Yes, because currently, we disabling only the possibility of following a user. 

If we need to extend functionality and hide followers from the list or etc I would recommend do it separately in follow-up and decide which AC should be applied, because it could make the process review of the task more complicated than was planned. Also, it will require additional testing.  
cc: @ribel